src/libraries/System.IO.FileSystem/src/System/IO/FileStatus.OSX.cs

```diff
@@ -0,0 +1,65 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+
+namespace System.IO
+{
+    internal partial struct FileStatus
+    {
+        internal unsafe void SetCreationTime(string path, DateTimeOffset time)
+        {
+            Interop.Sys.TimeSpec timeSpec = default;
+
+            long seconds = time.ToUnixTimeSeconds();
+
+            const long TicksPerMillisecond = 10000;
+            const long TicksPerSecond = TicksPerMillisecond * 1000;
+            long nanoseconds = (time.UtcDateTime.Ticks - DateTimeOffset.UnixEpoch.Ticks - seconds * TicksPerSecond) * NanosecondsPerTick;
```

Comment:
Nit, this could possibly be extracted into `private long UnixtimeSecondsToNanoseconds(long seconds)` to avoid duplicating with the existing code

Reply:
I'll put this in the next commit, I've implemented it locally with the signature `private static long UnixTimeSecondsToNanoseconds(DateTimeOffset time, long seconds)`.